http-core/src/main/scala/org/apache/pekko/http/javadsl/model/ContentType.scala

```diff
@@ -61,4 +61,10 @@ trait ContentType {
    * Returns the charset if this ContentType is non-binary.
    */
   def getCharsetOption: Optional[HttpCharset]
+
+  /**
+   * Returns the string representation of this ContentType
+   * @since 1.2.0
+   */
+  def value: String
 }
```

Comment:
I think this should be getValue. More Java API like. Other existing methods seem to be getters.

Reply:
Let me check because this isn't consistent, i.e. there is `getCharsetOption` but then there is also `mediaType` (which if it was consistent would be `getMediaType`)

Reply:
reasonable point - it does seem like the existing code can't decide on getter style versus field name style.

Reply:
In Spring-web, there is a `MimeTypeUtils#toString`, for me this method should be `getValue` or just override the `toString` method.

Reply:
You're right, the type is always String, toString is more in line with Java.

Reply:
toString would have the useful side effect that we would be ok to merge this to v1.1.x patch line

Reply:
I don't think putting `toString` on the interface is a good idea, its something that should concretely be implemented by its sub types.

Reply:
So fyi, changing it to `.getValue()` is more contrived than initially anticipated. Naively you would do something like

```scala
def getValue: String = value
```

However the `org.apache.pekko.http.javadsl.model.ContentType` has no visibility for the `value` method within the `ValueRenderable` trait (after all this is the exact issue we are dealing with). In order to make this work `org.apache.pekko.http.javadsl.model.ContentType` would have to extend `ValueRenderable` which would also expose the `value` method since its public i.e.
```
/**
 * INTERNAL API
 *
 * An entity that has a rendered value (like an HttpHeader)
 */
@InternalApi
private[http] trait ValueRenderable extends ToStringRenderable {
  def value: String = toString
}
```
which means the trait would both have `value` and `getValue` methods, far from ideal (note that even though the trait is marked as `private[http]`, this is only visible to the Scala compiler, Java has no concept of it and hence it only sees it as a standard interface with a public method).

I think this discussion is revealing of what appears to be an overlooked design misdecision, which is that we should be having 2 `ValueRenderable` traits, one for Java api and one for Scala api and only the respective implementations extending those traits.

I can go ahead and do this change but its obviously going to be more work. Thankfully `ValueRenderable` trait is marked as internal/private which means I only need to take care that a public `def value: String = toString` exists for the Scala api.

Wdyt?

Reply:
If leaving this as `def value = ???` is easier, I am ok with that. It seems like the Java API is already inconsistent stylistically.

Reply:
> If leaving this as `def value = ???` is easier, I am ok with that. It seems like the Java API is already inconsistent stylistically.

I personally would like to avoid that, I think that leaving unimplemented methods is taking it a bridge too far. @raboof what are your thoughts on this? I don't have an issue with doing this properly, this PR is slated for 1.2.0 anyways.

Reply:
I think it would make sense to keep this as `def value: String`. I don't think we need the `= ???` as all implementations of this trait have an implementation for this value, right?

Indeed 'getValue' would be more Java-esque, but since we already have `mediaType` this seems fine.

Reply:
Okay fair, I will mark this as resolved (if someone still disagrees then unresolve conversation).